### DISCUSSION TITLE
how to get PyCharm to detect installed libraries automatically?

### DISCUSSION THREAD

**Discussion by abdul-hamid-achik** [pdm-project/pdm#1258] [Q&A] [answered]
`Additionally, if you want to use tools from the environment (e.g. pytest), you have to add the __pypackages__/<major.minor>/bin directory to the PATH variable in the corresponding run/debug configuration.` from https://pdm.fming.dev/latest/usage/pep582/#pycharm does not seem to work as seen bellow

![Screen Shot 2022-07-21 at 17 50 13](https://user-images.githubusercontent.com/7379966/180327996-8a962e31-ba79-47e3-8421-db415e20d75e.png)

Pycharm does allow you to add a tool to the environment scope by adding a path variable for example https://www.jetbrains.com/help/pycharm/settings-path-variables.html#example

but it does not seem to work either

**Comment by frostming** [comment, marked as answer]
What do you mean by "not work"? Does it appear there but raises error when invoked? If so, what is the error?

**Reply by abdul-hamid-achik** [reply]
@frostming you know what? its indeed working I just thought it didn't because of the X but that seems to be a different issue, gonna close this one

I since then, ran pytest multipletimes with the `PATH=__pypackages__/*/bin` setting in the run config
